src/character.cpp

```diff
@@ -191,17 +193,24 @@ bool Character::move_effects(bool attacking)
             remove_effect("in_pit");
         }
     }
-    if (has_effect("grabbed")){
+    if (has_effect("grabbed") && !attacking){
         int zed_number = 0;
         for( auto &&dest : g->m.points_in_radius( pos(), 1, 0 ) ){
-            if (g->mon_at(dest) != -1){
+            if (g->mon_at(dest) != -1 && 
+                (
+                    g->zombie(g->mon_at(dest)).has_flag(MF_GRABS) ||
+                    g->zombie(g->mon_at(dest)).type->has_special_attack("GRAB")
+                )
+            ) {
                 zed_number ++;
             }
         }
-        if (attacking || zed_number == 0){
-            return true;
+        if ( zed_number == 0 ) {
+            add_msg_player_or_npc(m_good, _("You find yourself no longer grabbed."),
+                                            _("<npcname> finds themselves no longer grabbed."));
+            remove_effect("grabbed");
         }
-        if (get_dex() > get_str() ? rng(0, get_dex()) : rng( 0, get_str()) < rng( get_effect_int("grabbed") , 8) ){
+        else if (get_dex() > get_str() ? rng(0, get_dex()) : rng( 0, get_str()) < rng( get_effect_int("grabbed") , 8) ){
```

Comment:
you and the original coder both messed up the parentheses  :p

I'll fix it and comment it. There are actually a lot of conditions elsewhere in the code that could be replaced with max(dex,str). Now that I've got all the uses of dex and str documented, I'm probably going to go through all of them and audit for code readability and bugs.